src/main/java/uk/gov/companieshouse/controller/PaymentController.java

```diff
@@ -43,4 +49,30 @@ public PaymentGetResponse getPaymentUIData(@PathVariable("company-number") final
 
         return paymentService.get(dissolutionInfo.getETag(), companyNumber);
     }
+
+    @Operation(summary = "Patch Payment Status", tags = "Dissolution")
+    @ApiResponses(value = {
+            @ApiResponse(responseCode = "200", description = "Payment Status Updated"),
+            @ApiResponse(responseCode = "400", description = "Wrong status of Dissolution Application"),
+            @ApiResponse(responseCode = "404", description = "Dissolution Application not found")
+    })
+    @PatchMapping()
+    @ResponseStatus(HttpStatus.OK)
+    public void patchPaymentData(@PathVariable("company-number") final String companyNumber,
+                                 @Valid @RequestBody final PaymentPatchRequest body) {
+
+        logger.debug("[PATCH] Submitting payment data update request for company number {}", companyNumber);
+
+        DissolutionGetResponse dissolutionInfo = dissolutionService
+                .get(companyNumber)
+                .orElseThrow(DissolutionNotFoundException::new);
+
+        if (dissolutionInfo.getApplicationStatus() != ApplicationStatus.PENDING_PAYMENT) {
+            throw new DissolutionApplicationWrongStatusException();
+        }
+
+        if (PaymentStatus.PAID.equals(body.getStatus())) {
+            dissolutionService.updatePaymentStatus(body, companyNumber);
+        }
```

Comment:
Should we be validating the `state` (UUID) value here? Or will this be done on the frontend instead?

Reply:
State will be vaildated on the frontend as far as I know.